### PR TITLE
NSQ Kumar2024: freeze promoted comparison and stochastic-seed authority

### LINKED ISSUE #27: Kumar2024 preregistration: prospective calibration frontier and cohort-aware analysis
## Why preregister this study

Kumar2024 should be the first real longitudinal EEG evidence bundle for neurOS because it asks a deployment-relevant question: **how much target-session calibration is required after learning only from earlier sessions?**

Before running the public study at scale, freeze the analysis plan so dataset-specific choices cannot drift after seeing results.

## Upstream facts to preserve

Current upstream MOABB documents:

- 18 healthy BCI-naive participants;
- 6 sessions on separate days;
- MOABB session `0` is the offline calibration session;
- MOABB sessions `1`-`5` are the five online sessions in chronological order;
- session `0`: 4 bar-feedback runs = 80 trials = 40/class;
- session `1`: 4 online bar runs = 80 trials = 40/class;
- sessions `2`-`5`: 3 online bar runs = 60 trials = 30/class;
- 22 EEG channels, 512 Hz;
- left-hand vs right-hand motor imagery;
- original paper analysis band: 8-30 Hz;
- MOABB excludes car-racing runs and exposes the bar-feedback runs.

The current upstream loader constructs session keys in acquisition order: `0`, then `1` through `5`.

### Original treatment cohorts are not interchangeable

The neural trajectories were collected under two different adaptation/training protocols:

- MOABB subjects **1-9**: Generic Recentering (GR), unsupervised domain adaptation;
- MOABB subjects **10-18**: Personally Assisted Recentering (PAR), supervised recalibration.

This is a meaningful experimental covariate. Do not pool GR and PAR without preserving cohort identity in the evidence artifact and analysis.

## Primary prospective split

Use:

```text
history_policy = prior
```

For every participant:

```text
target session 1: train on 0
session 2:        train on 0,1
session 3:        train on 0,1,2
session 4:        train on 0,1,2,3
session 5:        train on 0,1,2,3,4
```

Future sessions are excluded.

This yields **18 subjects × 5 target sessions = 90 subject-session cases** before any data-quality failures.

Keep `all-other` as a separately labeled sensitivity analysis only. It is not prospective evidence.

## Calibration frontier

Primary labeled calibration budgets:

```text
0, 1, 2, 5, 10 examples / class
```

With a 0.5 fixed evaluation fraction, the nominal dataset structure supports 10/class in every online target session:

- MOABB session `1`: 40/class -> nominal 20/class calibration pool;
- sessions `2`-`5`: 30/class -> nominal 15/class calibration pool.

The strict paired-budget contract remains authoritative: if preprocessing/data issues reduce any actual case below 10/class, fail the promoted run rather than dropping that case from the right side of the curve.

## Evaluation split seeds

Do not choose the calibration/evaluation split after viewing model results.

### Pilot

Use one frozen seed only:

```text
2026
```

The pilot exists to validate data provenance, artifact semantics, runtime and feasibility. It is not the final model-comparison claim.

### Promoted comparison

Use a predeclared small ensemble of calibration/evaluation split seeds, with every competing method sharing each exact split fingerprint. Recommended initial set:

```text
2026, 3407, 9109
```

Report seed-level results and a seed-aggregated subject-session estimate. Do not search over these seeds.

## Initial transparent baseline

Method:

```text
8-30 Hz MOABB trials
-> CSP
-> LDA
```

This is the floor, not the neurOS claim.

Record exact CSP configuration, MNE/sklearn/MOABB versions, fit time and trial inference time.

## Method ladder after the floor is frozen

Under Issue #26, compare the exact same split/calibration identities across:

1. CSP + LDA;
2. EEGNet;
3. EEG-Conformer;
4. frozen representation + matched readout;
5. SourceWeigher transfer on one common frozen representation space;
6. future verified foundation encoders;
7. future ORION EEG representation only after an explicit EEG representation contract exists.

## SourceWeigher authority

For a target session, SourceWeigher may use only:

- embeddings from declared prior source sessions;
- embeddings from the declared target-session **calibration subset** at that budget.

It may not use final evaluation examples to estimate target moments.

At labeled budget `0`, target-dependent SourceWeigher is undefined unless a separate unlabeled-observation budget is explicitly introduced.

## Primary outcome

For method comparison, make the primary endpoint the **paired area under the calibration frontier** over the predeclared budgets `0,1,2,5,10` using balanced accuracy.

Why:

- rewards zero/few-shot usefulness rather than only the largest calibration budget;
- directly measures the recalibration burden neurOS/ORION are intended to reduce;
- uses the whole predeclared frontier;
- is harder to game with one favorable operating point.

Secondary outcomes:

- balanced accuracy at 0/class;
- balanced accuracy at 5/class;
- balanced accuracy at 10/class;
- ROC-AUC;
- Brier score / calibration error where probabilities are valid;
- fit/adaptation wall-clock;
- inference latency;
- failure rate;
- labels required to cross a predeclared performance threshold, once that threshold is chosen independently of the test results.

## Repeated-measures statistics

The 90 subject-session cases are **not 90 independent participants**.

Do not report naive trial/session-level confidence intervals as if every case were independent.

Recommended inference:

- paired method differences at the subject-session level;
- aggregate within subject before participant-level inference, or use a hierarchical/cluster bootstrap;
- preserve GR/PAR cohort as a stratification/covariate;
- report cohort-specific estimates in addition to the pooled estimate;
- report session-index effects because source-history size grows over time;
- for neural models, include predeclared model seeds and separate optimization variance from participant/session variance.

## Cohort-aware reporting

Every result row or linked case manifest should expose:

```text
subject
original_protocol = GR | PAR
held_out_session
source_sessions
history_policy
calibration_per_class
split_seed
partition_fingerprint
calibration_split_fingerprint
```

Primary public plots should offer:

- pooled frontier with cohort-aware uncertainty;
- GR frontier;
- PAR frontier;
- zero-calibration performance by target-session index.

This prevents a decoder improvement from being confused with the different online training histories of the two original experimental groups.

## Pilot execution

After PR #25 merges, first run a **small provenance pilot**, not the full headline study.

Recommended subjects:

```text
1,10
```

This deliberately includes one GR and one PAR participant.

Use:

```text
budgets = 0,1,2,5,10
history_policy = prior
seed = 2026
evaluation_fraction = 0.5
```

Inspect the complete evidence bundle manually before scaling.

Pilot acceptance criteria:

- observed sessions exactly match `0..5`;
- source sessions follow the prospective schedule above;
- cohort identity is preserved;
- all five budgets remain paired across all eligible cases;
- artifact hashes verify;
- no evaluation example enters preprocessing/adaptation fitting;
- runtime is practical enough to scale;
- report and raw CSV agree exactly.

## Scale-up gate

Only after the pilot passes:

1. freeze exact dependency/data identities;
2. run all 18 subjects for the CSP+LDA floor;
3. inspect failures and counts without changing the protocol to improve results;
4. freeze the benchmark bundle/split identities;
5. add EEGNet/EEG-Conformer/SourceWeigher under the same authority;
6. run the predeclared split/model seeds;
7. generate the Evidence Console directly from the artifact bundle.

## Claim boundary

This study can establish offline real-dataset evidence for prospective session transfer and recalibration burden.

It cannot by itself establish:

- live hardware reliability;
- online closed-loop benefit of neurOS;
- clinical benefit;
- ORION superiority unless ORION is actually evaluated under this frozen protocol;
- independence from the original GR/PAR experimental intervention.

### PR DESCRIPTION
**Pull request by sidhulyalkar**
## Why

The external-floor science is now mature enough to freeze the *comparison authority* before any promoted EEGNet final-assessment result is generated.

The preceding gates are durable:

- direct-upstream EEGNet training/state-selection authority merged through #95;
- Study Materialization Authority and Kumar bundle-v2 merged through #94;
- exact-main bundle-v2 classical artifact on `309b84f353d15e77172c50479c991478c8a78bcd` independently verified byte-for-byte after #96.

This PR implements the remaining preregistration/statistical authority from #27. It contains **no real-data execution loop** and does not authorize promoted EEGNet final-assessment execution by itself.

## Promoted comparison plan

Adds a content-addressed `Kumar2024ComparisonPlan` above individual `Kumar2024StudyConfig` realizations.

The frozen repeated-measures hierarchy is:

```text
participant
  -> target session
      -> split seed
          -> model seed, when applicable
              -> calibration budget
```

Only **participant** is an independent population unit. Sessions, split seeds, optimization seeds, and budgets never inflate N.

### Split authority

Predeclared split seeds:

```text
2026, 3407, 9109
```

Every competing method/model seed for a `(participant, target session, split seed)` must bind the same exact case-authority SHA or analysis fails closed.

### Optimization authority

CSP+LDA and RG+LR are deterministic and may not invent a model-seed axis.

EEGNet uses three predeclared optimization seeds:

```text
31415
384165836
3991196546
```

`31415` is the already-frozen #95 authority anchor. The other two were derived before promoted final-assessment execution from the first 32 bits of SHA-256 over:

```text
neuros.kumar2024.eegnet.optimization_seed.v1|1
neuros.kumar2024.eegnet.optimization_seed.v1|2
```

They are not selected from model performance.

The promoted bootstrap-analysis seed is independently SHA-derived as `160865088` and is distinct from split/model seeds.

## Complete-frontier inference

Primary pointwise metric remains `balanced_accuracy`.

Primary method-comparison endpoint is frozen as:

```text
paired_normalized_balanced_accuracy_frontier_auc
```

A model realization contributes only if every budget `0,1,2,5,10` succeeds. A subject/session/split contributes only if every required model seed is complete. A subject/session contributes only if every split seed is complete. A participant contributes only if every target session is complete. Pairwise inference uses only participants complete for both methods.

Missing rows, failed rows, missing model seeds, OOMs, and nonconvergence therefore cannot be averaged around.

## Validation before aggregation

The authority rejects:

- unplanned subjects, sessions, budgets, split seeds, model seeds, or methods;
- model seeds attached to deterministic methods;
- non-finite or out-of-range balanced accuracy;
- success without a score or failure carrying a score;
- duplicated realization/budget rows;
- GR/PAR cohort drift;
- case-authority mismatch across competing methods/model seeds.

## Stochastic-variance diagnostics

The primary inferential unit remains participant, but the analysis now preserves reviewer-facing descriptive layers:

1. complete model-realization frontier AUC;
2. model-seed-averaged participant/session/split AUC;
3. split-seed-averaged participant/session AUC;
4. participant-level estimate used for inference.

The first three are explicitly marked descriptive traceability only and never expose or increase an inferential participant count. If one EEGNet model-seed frontier fails, the corresponding split diagnostic is omitted rather than borrowing the other successful seed.

## Permanent CI ownership

The existing `NSQ Kumar2024 v1` authority workflow now:

- path-owns the comparison test file;
- runs the comparison authority suite on Python 3.10, 3.11, and 3.12;
- compiles the comparison module as part of the evidence surface.

No additional workflow is introduced.

## Validation already performed

Before branch cleanup, the exact comparison source/test blobs passed the focused synthetic authority gate together with the existing Kumar study and bundle-v2 tests. The branch was then rebuilt on latest durable `main` by transplanting those exact validated blobs, preserving the #96 CLI repair.

The final PR exact-head matrix is still required before merge.

## Explicit non-claims

This PR does not:

- run promoted EEGNet final-assessment results;
- claim EEGNet beats CSP or RG;
- qualify EEGConformer;
- introduce ORION;
- claim GR/PAR treatment effects;
- treat stochastic repeats as extra participants.

## Additional merge gate: training-only feasibility

The fully crossed promoted plan implies an upper design of 3 split seeds × 3 EEGNet model seeds × 90 subject/session cases × 5 calibration budgets = **4,050 EEGNet fits** before considering classical fits.

Before merging this authority as the final promoted plan, run a small **training-only** real-Kumar feasibility probe that constructs authorized fit sets and executes EEGNet fitting/internal validation only. It must not predict, score, or otherwise inspect untouched final-assessment observations.

If the full factorial is computationally practical, preserve this plan unchanged. If it is not, version a cheaper seed design *before* any promoted final-assessment result is exposed.

Tracks #27. Do not close #27 until the training-only feasibility gate confirms the final seed design.

### CODE REVIEW

**Comment by sidhulyalkar** [comment]
### Predeclared compute-feasibility decision rule

This is being frozen **before the fit-only probe produces a runtime result**. It must not be revised based on final-assessment performance, because the probe does not generate any.

For the exact frozen EEGNet recipe on the GitHub-hosted CPU reference environment:

- **slowest bracket fit <= 15 minutes**, with both bracket cases completing and no OOM/state-selection violation: preserve the full 3 split-seed x 3 model-seed comparison plan unchanged;
- **slowest bracket fit > 15 and <= 45 minutes**: do one additional fit-only GPU benchmark before freezing the promoted plan; do not inspect final assessment;
- **slowest bracket fit > 45 minutes**, OOM, or workflow timeout: the 3x3 factorial is not accepted as the promoted execution design. Version a cheaper seed plan before any promoted final-assessment result is generated.

The runtime evidence may report fit-set size, validation/state-selection metadata, epochs completed, tensor-state identity, and runtime. It may not call predict/predict_proba/score or serialize final-assessment X/y values.

This rule is a compute/sustainability gate only. Model accuracy is prohibited from influencing the seed-design decision.

**Comment by sidhulyalkar** [comment]
### Training-only feasibility gate: PASS, preserve the 3x3 plan unchanged

The predeclared compute rule was posted above before this result existed. The fit-only probe has now completed successfully without any final-assessment prediction/scoring path.

**Run:** `33223858335`
**Probe head:** `9444b0956e0e3f28786658568305170f538fb932`
**Artifact:** `kumar-eegnet-fit-feasibility-9444b0956e0e3f28786658568305170f538fb932`
**Artifact ID:** `9706393284`
**GitHub artifact digest:** `sha256:0537af4dbc52805525fbe49626ba6798f55ca5e4c14e353b7d32d3d998b27df6`

The artifact was downloaded and inspected independently.

#### Bracket fit A: light case
- case: subject 1 / target session 1 / split 2026 / budget 0
- authorized fit observations: `80`
- untouched assessment observations: `39`
- fit input: `80 x 22 x 2561`
- internal validation: `16` observations
- epochs completed / stopped epoch: `301 / 301`
- best observed validation epoch: `1`
- `.fit()` runtime: **115.812 s = 1.93 min**
- fitted state identity: tensor SHA-256
- final assessment used for state selection: **false**

#### Bracket fit B: heavy case
- case: subject 1 / target session 5 / split 2026 / budget 10/class
- authorized fit observations: `356`
- untouched assessment observations: `30`
- fit input: `356 x 22 x 2561`
- internal validation: `72` observations
- epochs completed / stopped epoch: `302 / 302`
- best observed validation epoch: `2`
- `.fit()` runtime: **453.294 s = 7.55 min**
- fitted state identity: tensor SHA-256
- final assessment used for state selection: **false**

#### Leakage boundary
Before either decoder was instantiated, the probe:
1. constructed the frozen longitudinal authority;
2. verified authorized fit indices were disjoint from untouched assessment indices;
3. copied only `X_fit/y_fit`;
4. deleted the participant-level data object;
5. called only `decoder.fit(...)` and `decoder.learned_state()`.

The serialized artifact contains fit/assessment index hashes and counts but **no assessment X/y, predictions, probabilities, accuracy, balanced accuracy, loss on assessment data, or other efficacy result**.

#### Compute decision
The predeclared rule preserves the full plan when the slowest bracket fit is <=15 minutes with no OOM/boundary failure. The observed slowest fit is **7.55 minutes**, so the rule requires us to **retain the 3 split-seed x 3 EEGNet-model-seed design unchanged**. No GPU adjudication is required.

The artifact projects approximately:
- `320.1` serial CPU-hours using the mean of the two bracket fits;
- `510.0` serial CPU-hours using the slower bracket fit for all 4,050 EEGNet fits.

This is substantial but tractable for a deliberately parallelized flagship study. It is not appropriate for routine CI.

No efficacy result has been generated. With the exact-head PR matrix already green, the additional compute gate is now satisfied.
